GitCommands/Git/GitModule.cs

```diff
@@ -52,10 +52,10 @@ public sealed partial class GitModule : IGitModule
         // 366dfba1abf6cb98d2934455713f3d190df2ba34 refs/tags/2.51
         //
         // Lines may also use \t as a column delimiter, such as output of "ls-remote --heads origin".
-        [GeneratedRegex(@"^(?<objectid>[0-9a-f]{40})[ \t](?<refname>.+)$", RegexOptions.Multiline | RegexOptions.ExplicitCapture)]
+        [GeneratedRegex(@"^(?<objectid>([0-9a-f]{40}|[0-9a-f]{64}))[ \t](?<refname>.+)$", RegexOptions.Multiline | RegexOptions.ExplicitCapture)]
```

Comment:
I wonder if this would avoid backtracking and give better performance:

```suggestion
        [GeneratedRegex(@"^(?<objectid>[0-9a-f]{40}([0-9a-f]{24})?)[ \t](?<refname>.+)$", RegexOptions.Multiline | RegexOptions.ExplicitCapture)]
```